packages/main/src/plugin/index.ts

```diff
@@ -1037,9 +1041,20 @@ export class PluginSystem {
         imageName: string,
         callbackId: number,
       ): Promise<void> => {
-        return containerProviderRegistry.pullImage(providerContainerConnectionInfo, imageName, (event: PullEvent) => {
-          this.getWebContentsSender().send('container-provider-registry:pullImage-onData', callbackId, event);
-        });
+        return progressImpl.withProgress<void>(
+          { location: ProgressLocation.TASK_WIDGET, title: `Pulling ${imageName}.` },
+          async progress => {
+            await containerProviderRegistry.pullImage(
+              providerContainerConnectionInfo,
+              imageName,
+              (event: PullEvent) => {
+                this.getWebContentsSender().send('container-provider-registry:pullImage-onData', callbackId, event);
+              },
+            );
+            progress.report({ increment: -1 });
```

Comment:
Not sure if it is really needed, I think it will just show undetermined progress anyway.